src/components/DateInput/Day.tsx

```diff
@@ -1,35 +1,105 @@
+import styled from '@emotion/styled';
 import dayjs, { Dayjs } from 'dayjs';
-import { noop } from '../../utils/helpers';
+import { noop, addOpacity } from '../../utils/helpers';
 
 export interface DayProps {
   day?: number;
+  isLastRow?: boolean;
   month: number;
   year: number;
   daySelected?: Dayjs;
+  dayRangeSelected?: [Dayjs?, Dayjs?];
   onDaySelect?: (date: Dayjs) => void;
 }
 
-const Cell: React.FC<BoxProps> = props => (
+const PseudoOuterCell: React.FC<BoxProps> = props => (
   <Box
     display="table-cell"
-    padding={2}
+    verticalAlign="middle"
     textAlign="center"
-    borderRadius="circle"
     cursor="pointer"
-    verticalAlign="middle"
-    _selected={{ backgroundColor: 'blue-400' }}
-    _hover={{ backgroundColor: 'blue-300' }}
+    position="relative"
     {...props}
   />
 );
 
-const MemoCell = React.memo(Cell);
+const OuterCell = styled(PseudoOuterCell)`
+  [aria-relevant]:hover {
+    background-color: ${({ theme }) => theme.colors['blue-300']};
+  }
+  [aria-placeholder] {
+    position: absolute;
+    z-index: 1;
+    width: 50%;
+    height: 100%;
+    top: 0;
+  }
+
+  &[aria-busy='true'] + [aria-selected='true'],
+  &[aria-selected='true'] + [aria-busy='true'],
+  &[aria-selected='true'] + &[aria-selected='true'] {
+    [aria-placeholder] {
+      background-color: ${({ theme }) => addOpacity(theme.colors['navyblue-100'], 0.2)};
+    }
+  }
+
+  &[aria-selected='true'] + &[aria-selected='true'] [aria-placeholder] {
+    width: 100%;
+    left: -50%;
+  }
+  &[aria-busy='true'] + &[aria-selected='true'] [aria-placeholder] {
+    left: 0;
+  }
+  &[aria-selected='true'] + &[aria-busy='true'] [aria-placeholder] {
+    left: -50%;
+  }
+
+  &[aria-selected='true'] [aria-relevant] {
+    background-color: ${({ theme }) => theme.colors['blue-400']};
+  }
+  &[aria-busy='true'] [aria-relevant] {
+    border-radius: 0;
+    background-color: ${({ theme }) => addOpacity(theme.colors['navyblue-100'], 0.2)};
+  }
+  &[aria-disabled='true'] + &[aria-busy='true'] {
+    [aria-relevant] {
+      border-radius: 4px 0 0 4px;
+    }
+  }
+  &[aria-busy='true'] {
+    &:last-of-type [aria-relevant] {
+      border-radius: 0 4px 4px 0;
+    }
+    &:first-of-type [aria-relevant] {
+      border-radius: 4px 0 0 4px;
+    }
+    &:only-of-type [aria-relevant] {
+      border-radius: 4px;
+    }
+  }
+`;
+
+const MemoOuterCell = React.memo(OuterCell);
+
+const PseudoCell: React.FC<BoxProps> = props => (
+  <Box padding={2} borderRadius="circle" position="relative" zIndex={2} {...props} />
+);
+
+const MemoCell = React.memo(PseudoCell);
 
-const Day: React.FC<DayProps> = ({ day, month, year, daySelected, onDaySelect = noop }) => {
+const Day: React.FC<DayProps> = ({
+  day,
+  month,
+  year,
+  daySelected,
+  isLastRow,
+  dayRangeSelected,
+  onDaySelect = noop,
+}) => {
   if (!day) {
-    return <MemoCell aria-disabled="true" />;
+    return isLastRow ? null : <MemoOuterCell aria-disabled="true" />;
```

Comment:
We need to ensure that there are now extra disabled cells as we want the pseudo CSS selectors take action, since this is a table the cells from the previous rows are holding everything in place. 
TL;DR This is used in order to make border-radius work

Reply:
@3nvi If a range is selected we need to highlight the cell between the given range. The first and the last cells of each row should get a tiny border-radius as seen here:
<img width="1076" alt="Screenshot 2020-09-01 at 4 57 24 PM" src="https://user-images.githubusercontent.com/1022166/91860525-74e8d980-ec74-11ea-9a01-d2567e16cc65.png">
 
This styling is applied using `:last-of-type` and `:first-of-type` pseudo-selectors, but in order to use the `:one-of-type` pseudo selector when there is just one cell in the last row we need to discard any disabled cells.

Reply:
> but in order to use the :one-of-type pseudo selector when there is just one cell in the last row we need to discard any disabled cells.

I think I understand it. Right now each month has some disabled dates (that belong to the next month) which are shown as disabled. In order for the CSS rule to apply you want to "hide" those days completely right?

Where  my  thought process fails  though, is that I see that you never  show any dates as disabled. When is this the case?

Reply:
I tried using `<Box _after />` but placeholders are affected by their siblings, thus we can't effectively style them.
Furthermore, styling gets even more bloated.